### ISSUE TITLE
Feature Request: [Feature] [Question] masking dynamic values, such as regex pattern matching

### ISSUE THREAD

**Issue by kwonoj** [fullstack-build/tslog#44]
**Description / Use Case for a Feature**

I may misunderstand docs and this is already supported, labeled question to clarify it.

There are some cases we would like to traverse all object payload recursively to detect a certain patterns of strings matching regex, such as prefix-based auth token and vice versa. Is there way to specify it via current `mask*` configuration values?

**Comment by terehov** [comment]
Hey @kwonoj, 

thank you for your feature request. I didn't quite get it. Could you please give me an example?

**Comment by kwonoj** [comment]
Our application has extensive redaction for any possible token logging point. We have some dynamic object payload, rough illustration like

```
{
 ...state,
 auth_token: 'token_prefix_...'
}
```

the important part is we don't have deterministic key to filter things out. above example, it was included in `auth_token`, but it could be other forms, including logger message itself. The only thing we can guarantee is token follows a specific form of pattern. 

Currently in our Winston, we do recursively traverse all input message& payload by overriding default formatted. Curious if there's similar way, or specify patterns other then known maskKey.

**Comment by terehov** [comment]
I think I got your point. The way I use it personally is by leveraging `maskAny` to hide any known keys. 
It was a design decision to avoid a recursive search in favor of RegEx for performance reasons. 
The way I understand your feature request is that you either don't want to tell `tslog` all possible "secrets", or maybe don't even have access to all of them. So I believe a plausible solution could be an additional method `maskAnyRegex`, where you could simply provide a RegEx that would be used for masking. 
Do you think that would work for your use case?

**Comment by kwonoj** [comment]
>  The way I use it personally is by leveraging maskAny to hide any known key

Yes, I agree with those design decision in general. I would say my usecase is more edge, aggressive cases if someone wants aggressive scanning even with cost of pattern matching since we can't leak either PII or tokens for security reasons.

> So I believe a plausible solution could be an additional method maskAnyRegex, where you could simply provide a RegEx that would be used for masking.

Yes, I think that could work.

Still, if I recall correctly this doesn't apply to message itself: is there way to enable matching against messages itself? as mentioned above, the way we currently do is via custom formatted - but doesn't seem like tslog provides way to override formatted for those usecase.

**Comment by kwonoj** [comment]
Thanks for the confirmation. So theoritically feature I would like to have is maskAny + regex (or predicate function).

**Comment by terehov** [comment]
Got it. 
Technically it was possible already since `maskAny` did accept RegEx. To make it more obvious I have released a version `v2.8.0` that reflects that in the docs as well as the naming of the setting `maskAnyRegEx`.

Here is an example for your use case:
```
const secretiveLogger = new Logger({
  name: "SecretiveLogger2",
  maskAnyRegEx: ["token_prefix_[\\w0-9_]*"],
});

secretiveLogger.info({
  x: "token_prefix_123 test1",
  y: "token_prefix_456_789 test2",
  z: "just a test",
});
```

**Comment by kwonoj** [comment]
Awesome, thanks for the help!
